image/image_linux.go

```diff
@@ -272,302 +273,228 @@ func makeLabel(now time.Time) string {
 	return now.UTC().Format("20060102")
 }
 
-func selectAssertionMaxFormats(tsto *tooling.ToolingStore, sysSn *seedwriter.SeedSnap) error {
-	if sysSn == nil {
-		// nothing to do
-		return nil
-	}
-	snapf, err := snapfile.Open(sysSn.Path)
-	if err != nil {
-		return err
-	}
-	// XXX take also kernel into account
-	// XXX warning logic
-	maxFormats, _, err := snap.SnapdAssertionMaxFormatsFromSnapFile(snapf)
-	if err != nil {
-		return err
-	}
-	tsto.SetAssertionMaxFormats(maxFormats)
-	return nil
-}
+type imageSeeder struct {
+	hasModes    bool
+	rootDir     string
+	bootRootDir string
+	seedDir     string
+	label       string
 
-var setupSeed = func(tsto *tooling.ToolingStore, model *asserts.Model, opts *Options) error {
-	if model.Classic() != opts.Classic {
-		return fmt.Errorf("internal error: classic model but classic mode not set")
-	}
-
-	hasModes := model.Grade() != asserts.ModelGradeUnset
-	var rootDir string
-	var bootRootDir string
-	var seedDir string
-	var label string
-	if !hasModes {
-		if opts.Classic {
-			// Classic, PrepareDir is the root dir itself
-			rootDir = opts.PrepareDir
-		} else {
-			// Core 16/18,  writing for the writeable partition
-			rootDir = filepath.Join(opts.PrepareDir, "image")
-			bootRootDir = rootDir
-		}
-		seedDir = dirs.SnapSeedDirUnder(rootDir)
-
-		// validity check target
-		if osutil.FileExists(dirs.SnapStateFileUnder(rootDir)) {
-			return fmt.Errorf("cannot prepare seed over existing system or an already booted image, detected state file %s", dirs.SnapStateFileUnder(rootDir))
-		}
-		if snaps, _ := filepath.Glob(filepath.Join(dirs.SnapBlobDirUnder(rootDir), "*.snap")); len(snaps) > 0 {
-			return fmt.Errorf("expected empty snap dir in rootdir, got: %v", snaps)
-		}
+	w     *seedwriter.Writer
+	model *asserts.Model
+	tsto  *tooling.ToolingStore
+}
 
+func (is *imageSeeder) modelessDirs(opts *Options) error {
```

Comment:
the name of this make it sound like it returns something but that is not the case: 

* setModelessDirs or initModelessDirs

Reply:
Nice suggestion, renamed to `setModelessDirs`